src/main/java/com/dougnoel/sentinel/steps/WindowAndTabSteps.java

```diff
@@ -70,18 +76,53 @@ public static void switchTo(String pageName) throws InterruptedException {
      * <li>I verify a new tab opens to the Google Maps page</li>
      * </ul>
      * @param pageName String the page to open
-     * @param direction String (next|previous) the window to open
+     * @param title String title of the page to go to
+     * @param direction String (next|previous) window to open
      * @throws InterruptedException if the thread gets interrupted
      */
-    @Then("I (?:am redirected to|remain on|switch to) the (.*?)(?: o?i?n the (previous|next) (?:window|tab))?$")
-    public static void switchTo(String pageName, String direction) throws InterruptedException {
+    @Then("I (?:am redirected to|remain on|switch to) the (.*?)(?:with the title \"([^\"]*)\")?(?: o?i?n the (previous|next) (?:window|tab))?$")
+    public static void switchTo(String pageName, String title, String direction) throws InterruptedException {
     	PageManager.setPage(pageName);
     	if (!StringUtils.isEmpty(direction)) {
-        	if (direction.equals("next"))
-        		Driver.goToNextWindow();
-        	else
-        		Driver.goToPreviousWindow();	
-    	}
+            if (direction.equals("next"))
+                Driver.goToNextWindow();
+            else if (direction.equals("previous"))
+                Driver.goToPreviousWindow();
+        }
+        else if (!StringUtils.isEmpty(title))
+            Driver.goToTitledWindow(title);
+
     	PageManager.waitForPageLoad();
     }
+
+    /**
+     * Waits until the title text is/contains the passed value
+     * @param comparisonType String (has|contains) the type of comparison to use
+     * @param titleText String the title to check for
+     * @return Boolean if the title was found
+     */
+    @Then("I wait until the (?:tab|window) (has|contains) the title text \"([^\"]*)\"$")
```

Comment:
Move implementation to Driver.
Make 2 methods: VerifyTabText, VerifyTabTextDoesNot
Call them below as a verification step

```suggestion
    @Then("I verify the (?:tab|window)( does not)? (has|have|contains?) the title text \"([^\"]*)\"$")
```

Move this method from WindowandTabSteps to Window and Tab Verification steps